src/device/arm/core_cortexm0.go

```diff
@@ -0,0 +1,5 @@
+// +build sam,atsamd21 nrf,nrf51
+
+package arm
+
+const Core = CortexM0
```

Comment:
Unfortunately, that's not entirely correct. The `atsamd21` is a Cortex-M0+ instead of a Cortex-M0. This is relevant because there are some small differences between the two cores (such as cycles per branch and privilege levels).

Perhaps more importantly, I do not think this is the right place for defining the `Core` constant. I think the right place is in the machine package in files for specific boards: that will reduce the number of places modifications need to be made to add support for a new board.

Thinking a bit longer about this, maybe simply providing a 'core' is not good enough:

* For fault status, what you're really interested in is the architecture (armv6m, armv7m, armv7em, armv8m.
* The runtime might be interested in which extensions are supported (such as floating point, or stack pointer limit checking in armv8m). It should preferably not need to know about the specific core.
* For timing sensitive operations, you need to know the exact core (to know the number of cycles an instruction takes).

Reply:
My motivation for putting the `Core` constant (and/or other constants) in the `arm` package is how it looks to the developer using TinyGo (vs devs working on TinyGo). I don't think a generic constant, e.g. `machine.CPU`, is the right choice for anything ARM-specific, so I would use `machine.ARMCore` etc. And IMO `arm.<Const>` (e.g. Core, Architecture, Extensions) flows better than `machine.ARM<Const>`, and better matches my expectations as a developer.

What about using `targets.json`? `llvm-target`, which is required, looks like it mostly matches architecture, and `cpu`, when it is used, seems to match the exact core. The only outlier is the GBA which defines `llvm-target` as `thumb4-none-eabi` and `cpu` as `arm7tdmi`. The compiler could use these, and possibly new `TargetSpec` fields, to (re)define constants.

Reply:
To expand on "better matches my expectations as a developer," I think it stems from the fact that the `machine` package is, in large part, a hardware abstraction layer, whereas the `device/*` packages provide low-level access to the microcontroller. So, "What architecture am I running on?" feels like far more of a low-level `device/*` kind of question than a `machine` hardware abstraction layer question.

One possibility is to create a `device/cpu` package and add the constants to that.